microservices/services/contest-service/src/main/java/org/xcolab/service/proposal/domain/proposal/ProposalDaoImpl.java

```diff
@@ -345,4 +363,109 @@ public Integer getProposalMaterializedPoints(Long proposalId) {
 
         return query.fetchInto(Proposal.class);
     }
+
+    public static void deleteOrphanProposals(DSLContext ctx) {
+        // Retrieve orphaned proposals
+        Select<Record1<Long>> orphanProposals = ctx.select(PROPOSAL.PROPOSAL_ID)
+                .from(PROPOSAL)
+                .whereNotExists(ctx.select(val(1))
+                        .from(PROPOSAL_2_PHASE)
+                        .where(PROPOSAL_2_PHASE.PROPOSAL_ID.eq(PROPOSAL.PROPOSAL_ID)));
+        // Delete proposal attributes of orphaned proposals
```

Comment:
Everything starting from here could be in a separate `deleteProposal` method.

Reply:
Done.